appbuilder/providers/appbuilder-livesync-provider-base.ts

```diff
@@ -0,0 +1,30 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+import Future = require("fibers/future");
+
+export abstract class AppBuilderLiveSyncProviderBase implements ILiveSyncProvider {
+	constructor(private $androidLiveSyncServiceLocator: {factory: Function},
+		private $iosLiveSyncServiceLocator: {factory: Function}) { }
+
+	public get platformSpecificLiveSyncServices(): IDictionary<any> {
+		return {
+			android: (_device: Mobile.IDevice, $injector: IInjector): IPlatformLiveSyncService => {
+				return $injector.resolve(this.$androidLiveSyncServiceLocator.factory, {_device: _device});
+			},
+			ios: (_device: Mobile.IDevice, $injector: IInjector) => {
+				return $injector.resolve(this.$iosLiveSyncServiceLocator.factory, {_device: _device});
+			}
+		};
+	}
+
+	public abstract buildForDevice(device: Mobile.IDevice): IFuture<string>;
```

Comment:
Maybe I'm wrong, but... `virtual` does not exist in TypeScript as all methods are "virtual".